src/lib.rs

```diff
@@ -7,31 +7,33 @@ extern crate quote;
 decl_derive!([Abomonation] => derive_abomonation);
 
 fn derive_abomonation(mut s: synstructure::Structure) -> quote::Tokens {
+
     let entomb = s.each(|bi| quote! {
-        ::abomonation::Abomonation::entomb(#bi, _writer);
+        ::abomonation::Abomonation::entomb(#bi, _write)?;
     });
 
-    s.bind_with(|_| synstructure::BindStyle::RefMut);
-    let embalm = s.each(|bi| quote! {
-        ::abomonation::Abomonation::embalm(#bi);
+    let extent = s.each(|bi| quote! {
+        sum += ::abomonation::Abomonation::extent(#bi);
     });
 
+    s.bind_with(|_| synstructure::BindStyle::RefMut);
+
     let exhume = s.each(|bi| quote! {
         let temp = bytes;
         let exhume_result = ::abomonation::Abomonation::exhume(#bi, temp);
```

Comment:
nit: Just write this as `bytes = ::abomonation::Abomonation::exhume(#bi, temp)?;`

Reply:
Probably. I recall that there is a tap dance to get the lifetimes to work out, and I wasn't sure if the issues would surface only at macro instantiation time, so I tried to do the minimal replacement.